### PR TITLE
Update Scalaz to 7.2.0

### PR DESCRIPTION
**Pull request by coltfred**
- Update scalaz to 7.2.0 and fix deprecation warnings surrounding Task.run, Task.timed and Task.attemptRun

### CODE REVIEW

**Comment by zaneli** [comment]
Personally, I welcome this PR.

ref #502

File: sbt [hunk 1]
```diff
@@ -1,132 +1,124 @@
 #!/usr/bin/env bash
 #
 # A more capable sbt runner, coincidentally also called sbt.
-# Author: Paul Phillips <paulp@typesafe.com>
```
Comment (pchlupacek) [inline sbt]: Are the changes in this file really mandatory for 7.2? If not, I just split them in another pr updating just the sbt script

File: src/test/scala/scalaz/stream/AsyncSignalSpec.scala [hunk 2]
```diff
@@ -138,7 +138,7 @@ class AsyncSignalSpec extends Properties("async.signal") {
       val feed = l.map(Some(_))
 
       val ref = async.signalUnset[Option[Int]]
-      ref.set(initial).run
+      ref.set(initial)unsafePerformSync
```
Comment (pchlupacek) [inline src/test/scala/scalaz/stream/AsyncSignalSpec.scala]: isn't .dot missing there ?

File: src/test/scala/scalaz/stream/AsyncTopicSpec.scala [hunk 3]
```diff
@@ -95,15 +95,15 @@ class AsyncTopicSpec extends Properties("topic") {
     l: List[Int] =>
       (l.size > 0 && l.size < 10000) ==> {
         val topic = async.topic[Int]()
-        topic.fail(Bwahahaa).run
+        topic.fail(Bwahahaa)unsafePerformSync
```
Comment (pchlupacek) [inline src/test/scala/scalaz/stream/AsyncTopicSpec.scala]: please put the dot after fail

File: src/test/scala/scalaz/stream/AsyncTopicSpec.scala [hunk 4]
```diff
@@ -95,15 +95,15 @@ class AsyncTopicSpec extends Properties("topic") {
...
 
 
         val emitted = new SyncVar[Throwable \/ Unit]
-        Task((Process.emitAll(l).toSource to topic.publish).run.runAsync(emitted.put)).run
+        Task((Process.emitAll(l).toSource to topic.publish).run.unsafePerformAsync(emitted.put))unsafePerformSync
```
Comment (pchlupacek) [inline src/test/scala/scalaz/stream/AsyncTopicSpec.scala]: please put the dot

File: src/test/scala/scalaz/stream/NioSpec.scala [hunk 5]
```diff
@@ -203,13 +203,13 @@ class NioSpec extends Properties("nio") {
...
-      .runAsync(clientGot.put)
-    ).run
+      .unsafePerformAsync(clientGot.put)
+    )unsafePerformSync
 
     clientGot.get(6000)
     serverGot.get(6000)
 
-    stop.set(true).run
+    stop.set(true)unsafePerformSync
```
Comment (pchlupacek) [inline src/test/scala/scalaz/stream/NioSpec.scala]: dot please. can you please assures dot are putted everywhere when chaining the calls?

**Comment by pchlupacek** [comment]
I have feeling something must have changed about the Task/Future implementation that broke the tests. Can you please identify which of the tests is hung exactly? Perhaps then we can see more closely where the problem is.

File: sbt (hunk 1 above)
Comment (coltfred) [inline sbt]: They're required to make the build run at all. The SBT url was incorrect. I can open it as a separate PR on this branch if that's preferable.

**Comment by coltfred** [comment]
Opened #532 which has the same logical changes, but without fixing the deprecation warnings.

**Comment by coltfred** [comment]
We decided that #532 was better because it has less changes. Closing this.
